### PR TITLE
REL: set version to 1.16.0.dev0

### PR DESCRIPTION
**Pull request by tylerjereddy**
* Don't merge until the `1.15.0` release notes updates have been merged and the `maintenance/1.15.x` branch has been pushed up (I'll probably merge when the time is right).

* The changed dependency versions in the `1.16.0` release notes are just drafts/subject to change. Note that I've not taken on the task of actually updating those dependencies in places where they may get checked programmatically (i.e., `pyproject.toml`), since that may be a larger project that can happen after branching.

* It looks like our version handling has actually simplified since the previous release--see the matching PR from previous release at gh-20827.

[docs only]

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
Thanks @tylerjereddy. I reminded myself of the version handling change - this indeed looks as expected.

**Comment by tylerjereddy** [comment]
The release notes have been merged, the `maintenance/1.15.x` branch has been pushed up, there's a core dev approval and CI is passing, so `main` branch is ready to accept changes for the `1.16.x` series.

**Comment by tylerjereddy** [comment]
Note for future self--Matt improved the way the section subheaders link to the TOC in the cross-listed PR, so when making future relnotes from scratch try to follow that format.

**Pull request merged**
